src/mono/browser/runtime/runtime.c

```diff
@@ -209,17 +235,13 @@ get_native_to_interp (MonoMethod *method, void *extra_arg)
 	const char *class_name = mono_class_get_name (klass);
 	const char *method_name = mono_method_get_name (method);
 	char key [128];
+	char fixedName[256];
```

Comment:
If `key` has max length of 128, do we need double size from `fixedName`? Should both of them be `256`?
What happens when the `key` is longer?

Reply:
Alos, can we write it in a way that allocates only one buffer?

Reply:
In `fixedName` for some cases will be mapping to 4 characters https://github.com/dotnet/runtime/pull/101106/files#diff-6a06405bac4c9b4b5bb48fef0095abb9f1b233118903560c07f93e7f5b09658eR208 , that's why I decided to take double size.
I will try to see if it can be done using one buffer.

Reply:
> In `fixedName` for some cases will be mapping to 4 characters https://github.com/dotnet/runtime/pull/101106/files#diff-6a06405bac4c9b4b5bb48fef0095abb9f1b233118903560c07f93e7f5b09658eR208 , that's why I decided to take double

Oh, thanks! I missed that

Reply:
I tried refactoring to use one buffer and it requires shifting characters for every hexadecimal representation insertion. I decided to keep fixedName here, as it can cause performance overhead, especially for large strings.